sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultResultParameterProcessor.java

```diff
@@ -239,106 +239,105 @@ protected BArray createAndPopulateCustomBBRefValueArray(Object firstNonNullEleme
-                boolDataArray.add(i, ((Boolean) dataArray[i]).booleanValue());
-            }
-            return boolDataArray;
-        } else if (firstNonNullElement instanceof Short) {            
-            BArray shortDataArray = ValueCreator.createArrayValue(intArrayType);
-            for (int i = 0; i < length; i++) {
-                shortDataArray.add(i, ((Short) dataArray[i]).intValue());
-            }
-            return shortDataArray;
-        } else if (firstNonNullElement instanceof Integer) {            
-            BArray intDataArray = ValueCreator.createArrayValue(intArrayType);
-            for (int i = 0; i < length; i++) {
-                intDataArray.add(i, ((Integer) dataArray[i]).intValue());
-            }
-            return intDataArray;
-        } else if (firstNonNullElement instanceof Long) {
-            BArray longDataArray = ValueCreator.createArrayValue(intArrayType);
-            for (int i = 0; i < length; i++) {
-                longDataArray.add(i, ((Long) dataArray[i]).longValue());
-            }
-            return longDataArray;
-        } else if (firstNonNullElement instanceof Float) {
-            BArray floatDataArray = ValueCreator.createArrayValue(floatArrayType);
-            for (int i = 0; i < length; i++) {
-                floatDataArray.add(i, ((Float) dataArray[i]).floatValue());
-            }
-            return floatDataArray;
-        } else if (firstNonNullElement instanceof Double) {
-            BArray doubleDataArray = ValueCreator.createArrayValue(floatArrayType);
-            for (int i = 0; i < dataArray.length; i++) {
-                doubleDataArray.add(i, ((Double) dataArray[i]).doubleValue());
-            }
-            return doubleDataArray;
-        } else if (firstNonNullElement instanceof BigDecimal) {
-            BArray decimalDataArray = ValueCreator.createArrayValue(decimalArrayType);
-            for (int i = 0; i < dataArray.length; i++) {
-                decimalDataArray.add(i, ValueCreator.createDecimalValue((BigDecimal) dataArray[i]));
-            }
-            return decimalDataArray;
-        } else if (firstNonNullElement instanceof byte[]) {            
-            BArray byteDataArray = ValueCreator.createArrayValue(byteArrayType);
-            for (int i = 0; i < dataArray.length; i++) {                
-                byteDataArray.add(i, ValueCreator.createArrayValue((byte[]) dataArray[i]));
-            }
-            return byteDataArray;
-        } else if (firstNonNullElement instanceof java.util.Date) {            
-            BArray mapDataArray;
-            if (firstNonNullElement instanceof Date) {
+        String elementType = firstNonNullElement.getClass().getCanonicalName();
+        switch (elementType) {
+            case Constants.Classes.STRING:
+                BArray stringDataArray = ValueCreator.createArrayValue(stringArrayType);
+                for (int i = 0; i < length; i++) {
```

Comment:
Yes, we can remove the code duplication by using this. But is this an effective way? bcoz we need to check the type every time.

Reply:
I think regarding the performance, there won't be any difference since we assign the type to a variable at the beginning (`String elementType = firstNonNullElement.getClass().getCanonicalName();`). But yes, there are pros and cons to this approach. 
pros:

1. Can collapse switch cases for some types (eg: `SHORT`, `INTEGER`, `LONG`) during array creation.
2. Can populate the array using a single for loop irrespective of the type.

cons:

1. Need to check the type when creating the array as well as populating the array.

But I'm +1 to keep the code unchanged. :)